python/cuda_parallel/cuda/parallel/experimental/_utils/protocols.py

```diff
@@ -61,3 +61,31 @@ def is_contiguous(arr: DeviceArrayLike) -> bool:
     else:
         # not contiguous
         return False
+
+
+def _validate_and_get_stream(stream) -> Optional[int]:
```

Comment:
Nit: since this function is being used outside of this module, it should be "public"
```suggestion
def validate_and_get_stream(stream) -> Optional[int]:
```